File: Program/ActEditor/bin/pl/wroc/pwr/student/acteditor/parsing/xsd/XSDHelper.js
Clazz.declarePackage ("pl.wroc.pwr.student.acteditor.parsing.xsd");
Clazz.load (null, "pl.wroc.pwr.student.acteditor.parsing.xsd.XSDHelper", ["pl.wroc.pwr.student.acteditor.model.tags.AttributeGroup", "$.Composition", "$.SimpleAttribute", "$.SimpleType"], function () {
c$ = Clazz.declareType (pl.wroc.pwr.student.acteditor.parsing.xsd, "XSDHelper");
Clazz.makeConstructor (c$, 
function () {
});
Clazz.defineMethod (c$, "createSimpleTypeDefinition", 
function (line) {
var simpleType =  new pl.wroc.pwr.student.acteditor.model.tags.SimpleType ();
var name = this.getAttribute ("name", line);
simpleType.setName (name);
return simpleType;
}, "~S");
Clazz.defineMethod (c$, "createGroupReference", 
function (line) {
var attribute =  new pl.wroc.pwr.student.acteditor.model.tags.AttributeGroup ();
var name = this.getAttribute ("ref", line);
attribute.setName (name);
return attribute;
}, "~S");
Clazz.defineMethod (c$, "createGroupDefinition", 
function (line) {
var attribute =  new pl.wroc.pwr.student.acteditor.model.tags.AttributeGroup ();
var name = this.getAttribute ("name", line);
attribute.setName (name);
return attribute;
}, "~S");
Clazz.defineMethod (c$, "createAttribute", 
function (line) {
var name = this.getAttribute ("name", line);
var defaultValue = this.getAttribute ("default", line);
var type = this.getAttribute ("type", line);
var use = this.getAttribute ("use", line);
var attribute =  new pl.wroc.pwr.student.acteditor.model.tags.SimpleAttribute ();
attribute.setName (name);
attribute.setType (type);
if (!defaultValue.equals ("")) {
attribute.setDefault (defaultValue);
}if (!use.equals ("")) {
attribute.setUse (use);
}return attribute;
}, "~S");
Clazz.defineMethod (c$, "createElementGroupReference", 
function (line) {
var name = this.getAttribute ("ref", line);
var element =  new pl.wroc.pwr.student.acteditor.model.tags.Composition (name, "ref");
return element;
}, "~S");
Clazz.defineMethod (c$, "createElementGroupDefinition", 
function (line) {
var name = this.getAttribute ("name", line);
var element =  new pl.wroc.pwr.student.acteditor.model.tags.Composition (name, "group");
return element;
}, "~S");
Clazz.defineMethod (c$, "createElementReference", 
function (line) {
var name = this.getAttribute ("ref", line);
var minOccurs = this.getAttribute ("minOccurs", line);
var maxOccurs = this.getAttribute ("maxOccurs", line);
var element =  new pl.wroc.pwr.student.acteditor.model.tags.Composition (name, "ref");
if (!minOccurs.equals ("")) {
element.setMinOccurs (minOccurs);
}if (!maxOccurs.equals ("")) {
element.setMaxOccurs (maxOccurs);
}return element;
}, "~S");
Clazz.defineMethod (c$, "setDescription", 
function (object, line) {
var description = line.substring (line.indexOf (">") + 1, line.lastIndexOf ("<"));
if (Clazz.instanceOf (object, pl.wroc.pwr.student.acteditor.model.tags.Composition) || Clazz.instanceOf (object, pl.wroc.pwr.student.acteditor.model.tags.SimpleElement)) {
if ((object).getDescription ().length == 0) {
(object).setDescription (description);
}} else if (Clazz.instanceOf (object, pl.wroc.pwr.student.acteditor.model.tags.Attribute) || Clazz.instanceOf (object, pl.wroc.pwr.student.acteditor.model.tags.AttributeGroup)) {
if ((object).getDescription ().length == 0) {
(object).setDescription (description);
}} else if (Clazz.instanceOf (object, pl.wroc.pwr.student.acteditor.model.tags.SimpleType)) {
if ((object).getDescription ().length == 0) {
(object).setDescription (description);
}}return object;
}, "~O,~S");
Clazz.defineMethod (c$, "createElementDefinition", 
function (line) {
var name = this.getAttribute ("name", line);
var result =  new pl.wroc.pwr.student.acteditor.model.tags.Composition (name, "all");
return result;
}, "~S");
Clazz.defineMethod (c$, "getAttribute", 
function (name, line) {
if (line.contains (name)) {
var begin = line.indexOf (name + "=\"") + (name + "=\"").length;
var end = line.indexOf ("\"", line.indexOf (name + "=\"") + (name + "=\"").length);
return line.substring (begin, end);
}return "";
}, "~S,~S");
});

File: Program/ActEditor/src/pl/wroc/pwr/student/acteditor/parsing/xsd/XSDHelper.java
package pl.wroc.pwr.student.acteditor.parsing.xsd;

import java.util.Stack;

import pl.wroc.pwr.student.acteditor.model.AttributeRegistry;
import pl.wroc.pwr.student.acteditor.model.ElementRegistry;
import pl.wroc.pwr.student.acteditor.model.TypeRegistry;
import pl.wroc.pwr.student.acteditor.model.tags.Attribute;
import pl.wroc.pwr.student.acteditor.model.tags.AttributeGroup;
import pl.wroc.pwr.student.acteditor.model.tags.Composition;
import pl.wroc.pwr.student.acteditor.model.tags.Element;
import pl.wroc.pwr.student.acteditor.model.tags.SimpleAttribute;
import pl.wroc.pwr.student.acteditor.model.tags.SimpleElement;
import pl.wroc.pwr.student.acteditor.model.tags.SimpleType;

/**
 * Klasa pomocnicza do wczytywania dokumentu XML.
 * 
 * @author Mateusz
 *
 */
public class XSDHelper {
	
	public XSDHelper() {
	}

	public SimpleType createSimpleTypeDefinition(String line) {
		SimpleType simpleType = new SimpleType();
		String name = getAttribute("name", line);
		simpleType.setName(name);
		return simpleType;
	}

	public Attribute createGroupReference(String line) {
		Attribute attribute = new AttributeGroup();
		String name = getAttribute("ref", line);
		attribute.setName(name);
		return attribute;
	}

	public Attribute createGroupDefinition(String line) {
		Attribute attribute = new AttributeGroup();
		String name = getAttribute("name", line);
		attribute.setName(name);
		return attribute;
	}

	public Attribute createAttribute(String line) {
		String name = getAttribute("name", line);
		String defaultValue = getAttribute("default", line);
		String type = getAttribute("type", line);
		String use = getAttribute("use", line);
		Attribute attribute = new SimpleAttribute();
		attribute.setName(name);
		attribute.setType(type);
		if(!defaultValue.equals("")) {
			attribute.setDefault(defaultValue);
		}
		if(!use.equals("")) {
			attribute.setUse(use);
		}
		return attribute;
	}

	public Element createElementGroupReference(String line) {
		String name = getAttribute("ref", line);
		Element element = new Composition(name, "ref");
		return element;
	}

	public Element createElementGroupDefinition(String line) {
		String name = getAttribute("name", line);
		Element element = new Composition(name, "group");
		return element;
	}

	public Element createElementReference(String line) {
		String name = getAttribute("ref", line);
		String minOccurs = getAttribute("minOccurs", line);
		String maxOccurs = getAttribute("maxOccurs", line);
		Element element = new Composition(name, "ref");
		if(!minOccurs.equals("")) {
			element.setMinOccurs(minOccurs);
		}
		if(!maxOccurs.equals("")) {
			element.setMaxOccurs(maxOccurs);
		}
		return element;
	}

	public Object setDescription(Object object, String line) {
		String description = line.substring(line.indexOf(">")+1, line.lastIndexOf("<"));
		if(object instanceof Composition || object instanceof SimpleElement) {
			if(((Element)object).getDescription().length() == 0) {
				((Element)object).setDescription(description);
			}
		} else if(object instanceof Attribute || object instanceof AttributeGroup) {
			if(((Attribute)object).getDescription().length() == 0) {
				((Attribute)object).setDescription(description);
			}
		} else if(object instanceof SimpleType) {
			if(((SimpleType)object).getDescription().length() == 0) {
				((SimpleType)object).setDescription(description);
			}
		}
		return object;
	}

	public Element createElementDefinition(String line) {
		String name = getAttribute("name", line);
		Element result = new Composition(name, "all");
		return result;
	}

	public String getAttribute(String name, String line) {
		if(line.contains(name)) {
			int begin = line.indexOf(name + "=\"") + (name + "=\"").length();
			int end = line.indexOf("\"", line.indexOf(name + "=\"") + (name + "=\"").length());
			return line.substring(begin, end);
		}
		return "";
	}
}
